paraclear/get_balance_onchain.py

```diff
@@ -0,0 +1,228 @@
+            if re.search(err_msg, err.message, re.IGNORECASE) or err.code == RPC_CONTRACT_ERROR:
+                return None
+            raise err
+        return implementation
+
+    @staticmethod
+    def _get_implementation_call(address: Address) -> Call:
+        return Call(
+            to_addr=address,
+            selector=get_selector_from_name("implementation"),
+            calldata=[],
+        )
+
+def as_int(value: int) -> int:
+    """
+    Returns the lift of the given field element, val, as an integer
+    in the range (-prime/2, prime/2).
+    """
+    return value if value < FIELD_PRIME // 2 else value - FIELD_PRIME
+
+
+def to_decimal(value_felt: int, decimals=PARACLEAR_QUANTUMS):
+    int_value = as_int(value_felt)
+    output = Decimal(int_value).scaleb(-decimals)
+    return output
+
+def int_16(val):
+    if isinstance(val, str):
+        return int(val, 16)
+    elif isinstance(val, int):
+        return val
+    else:
+        raise ValueError(f"Expected string or integer, got {type(val)}")
+
+
+def get_client(node_url: str = FULL_NODE_URL):
+    return FullNodeClient(node_url=node_url)
+
+
+def get_proxy_config():
+    return ProxyConfig(
+        max_steps=5,
+        proxy_checks=[StarkwareETHProxyCheck(), ArgentProxyCheck(), OpenZeppelinProxyCheck()],
+    )
+
+
+async def load_contract(
+    address: str,
+    client: FullNodeClient,
+    proxy_config: Union[bool, ProxyConfig] = get_proxy_config(),
+) -> Contract:
+    return await Contract.from_address(
+        address=address, provider=client, proxy_config=proxy_config
+    )
+
+def get_markets_from_api():
+    headers = {"Accept": "application/json"}
+    try:
+        resp = requests.get(f"{PRDX_API_URL}/markets", headers=headers)
+        
+        # Raise an exception for bad status codes
+        resp.raise_for_status()
+        
+        # Try to parse JSON
+        try:
+            data = resp.json()
+        except requests.exceptions.JSONDecodeError as e:
+            print(f"JSONDecodeError: {e}")
+            print(f"Response content: {resp.text}")
+            return []
+
+        if "results" in data:
+            return [r["symbol"] for r in data["results"]]
+        else:
+            print("Expected 'results' key not found in API response")
+            print(f"Response data: {data}")
+            return []
+    except requests.exceptions.RequestException as e:
+        print(f"Request failed: {e}")
+        return []
+
+
+async def get_balance(target_address: str):
+    print("Loading Client...")
+    target_address_int = int_16(target_address)
+
+    client = get_client(node_url=FULL_NODE_URL)
+    print("Client loaded")
+    print("Loading Paraclear contract...")
+    paraclear_contract = await load_contract(PARACLEAR_ADDRESS, client=client)
+    print("Paraclear contract loaded")
+    print("Getting USDC contract...")
+    usdc_contract = await load_contract(USDC_TOKEN_ADDRESS, client=client)
+    print("USDC contract loaded")
```

Comment:
Why did you add this back?   We don't actually use the variable `usdc_contract` in this code.   We do need the env var so that we can query paraclear properly but I think we can keep this out of the code.